sdk/python/feast/feature_store.py

```diff
@@ -1906,6 +2169,17 @@ def serve(
         registry_ttl_sec: int = 2,
     ) -> None:
         """Start the feature consumption server locally on a given port."""
+        print("Start Prometheus Server")
```

Comment:
I think this should be separated from feature server . The metrics server should be start separately then feature server. Currently that is the part of feature server command.
you can add new command `feast serve_metrics` which will run the metrics endpoint

Reply:
Not sure how that would work, that would be a separate process, wouldn't it?

Reply:
Yes, will be separate processes as the code starting Prometheus server so I am suggesting to create independent server and users can use the command `feast serve_metrics` to start the server and then be accessed via http://localhost:port/metrics

Reply:
I'm not really a prometheus user myself, but my understanding is the way the client library works is that whatever metrics you send (or set) throughout your application code is picked up and exposed by the prometheus server running in the same process. If an app code and prometheus server are running in different processes, there'd be no way to communicate in order to set metrics. Or maybe I'm misreading the whole thing.

Reply:
I got your point, yes it should be in same process to collect the metrics and agree to go with @tchughesiv suggestion.